src/PhpSpec/Console/Provider/NamespacesAutocompleteProvider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Console\Provider;
+
+use Symfony\Component\Finder\Finder;
+
+final class NamespacesAutocompleteProvider
```

Comment:
`NamespacesAutocompleteProvider` was just a quick guess of possible name for that class, so feel free to change it if more suitable name can be found :smile:

Reply:
This can now be replaced by the namespace provider introduced in #1122 - it isn't currently registered as a service but it probably should be!

Reply:
actually you know what - this is different to the namespace provider concept used for the composer autoconfiguration - what you have is actually fine.